### PR TITLE
TEST: add more tests for deleteBucket with versioning

### PR DESCRIPTION
**Pull request by nicolas2bert**
Testing delete bucket
- if delete marker still bucket
- if version still in bucket
- if no versions or delete marker still in bucket

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/versioning/bucketDelete.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+const testing = process.env.VERSIONING === 'no' ? describe.skip : describe;
+
+function checkError(err, code) {
+    assert.notEqual(err, null, 'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+}
+
+function checkNoError(err) {
+    assert.equal(err, null,
```
Comment (jonathan-gramain) [inline tests/functional/aws-node-sdk/test/versioning/bucketDelete.js]: maybe use `assert.ifError` instead

File: tests/functional/aws-node-sdk/test/versioning/bucketDelete.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                if (err && err.code === 'NoSuchBucket') {
+                    return done();
+                } else if (err) {
+                    return done(err);
+                }
+                return s3.deleteBucket({ Bucket: bucketName }, done);
+            });
+        });
+
+        it('should be able to delete empty bucket with version unabled',
```
Comment (jonathan-gramain) [inline tests/functional/aws-node-sdk/test/versioning/bucketDelete.js]: unabled -> enabled

Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/bucketDelete.js]: thanks :)

**Comment by nicolas2bert** [comment]
@ironman-machine try

**Pull request merged**